src/app/card/card.components.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Diary } from '../model/diary';
+
+@Component({
+  selector: 'app-card',
+  templateUrl: './card.component.html',
+  styleUrls: ['./card.component.scss'],
+})
+export class CardComponent implements OnInit {
+  @Input() id: string = '';
```

Comment:
don't have to use this many inputs. Make it object then you only have to use the one @Input()

Reply:
done